lib/parser.js

```diff
@@ -31,12 +31,12 @@ module.exports = {
 /**
  * Parses and validate an AsyncAPI document from YAML or JSON.
  * 
- * @param {String} asyncapiYAMLorJSON An AsyncAPI document in JSON or YAML format.
- * @param {Object} [options] Configuration options.
- * @param {String} [options.path] Path to the AsyncAPI document. It will be used to resolve relative references. Defaults to current working dir.
- * @param {Object} [options.parse] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
- * @param {Object} [options.resolve] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
- * @param {Object} [options.applyTraits=true] Whether to resolve and apply traits or not.
+ * @param {string} asyncapiYAMLorJSON An AsyncAPI document in JSON or YAML format.
+ * @param {object} [options] Configuration options.
+ * @param {string} [options.path] Path to the AsyncAPI document. It will be used to resolve relative references. Defaults to current working dir.
+ * @param {object} [options.parse] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
+ * @param {object} [options.resolve] Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
+ * @param {object} [options.applyTraits=true] Whether to resolve and apply traits or not.
```

Comment:
This section seems to be in conflict too. It appears to be the same fields and documentation of the type `ParserOptions` defined here:
https://github.com/asyncapi/parser-js/blob/83e9d4ea774255d71d8aa4594c4a29b02eb96062/lib/parser.js#L32-L37

What do you think about keeping the initial documentation and use the defined `ParserOptions`? Also, this type could be used to document the `options` parameter of the function `dereference`, it seems to have the same format.